### DISCUSSION TITLE
Style Selector: Constrain on (DataContext) type

### DISCUSSION THREAD

**Discussion by pr8x** [AvaloniaUI/Avalonia#7741] [Ideas]
I've been wanting a feature like for a while now: 

`<Style Selector="ListBoxItem[DataContext~vms:ISimpleItemViewModel]" >`

Syntax is debatable, but having some general-purpose operator for checking type equality (not just value equality) of properties would be certainly nice to have. There are many cases (in my experience) where you want to style some container based on the VM it hosts. Currently it's required to overwrite `ItemContainerGenerator` and create different containers for different VMs (or by exposing some `bool` on the VM and bind it to `Classses.my-custom-view`)

EDIT: For clarity this is not just about `DataContext`. The operator could be used for other properties as well: 

`<Style Selector="MyControl[FruitType~vms|Apple]" />`

**Comment by pr8x** [comment]
CompiledBindings will actually throw if the `x:DataType` provided on a style does not match the `DataContext` of the item. I wonder if we could just ignore these styles and treat the `DataType` attribute as part of the style constraint? Or go with the proposed solution of allowing constraints on type.

**Comment by Gamer-XP** [comment]
Was a way to do this added since 2022?
